### PR TITLE
codec(ticdc): simple protocol decoder parse timestamp with microsecond precision

### LINKED ISSUE #10703: simple protocol decoder parse timestamp value with microsecond precision
as title

### LINKED ISSUE #10705: kafka consumer panic due to decode message value fail
### What did you do?

1. Create kafka changefeed with row checksum on
```
`[integrity]
integrity-check-level = "correctness"
corruption-handle-level = "error"`
```
2. run workload which includes timestamp columns, whose value like "2024-03-01 13:40:39.662659"
3. run a kafka consumer to consume the Kafka message received from CDC, and validatation of row checksum is endable for the the consumer to validate the integrity of the message.

### What did you expect to see?

kafka consumer row checksum valid should pass

### What did you see instead?

CDC doesn't report error for row checksum, but kafka consumer reported error:

```
[2024/03/01 13:40:41.262 +00:00] [INFO] [decoder.go:372] ["table info stored"] [schema=asim] [table=t2] [version=448082059088887827]
[2024/03/01 13:40:41.270 +00:00] [INFO] [tz.go:34] ["Use the timezone of the TiCDC server machine"] [timezoneName=System] [timezone=UTC]
[2024/03/01 13:40:41.270 +00:00] [ERROR] [verify_checksum.go:39] ["checksum mismatch"] [expected=803306644] [actual=3014876872]
[2024/03/01 13:40:41.270 +00:00] [INFO] [message.go:429] ["checksum corrupted on the previous columns"]
[2024/03/01 13:40:41.270 +00:00] [WARN] [message.go:457] ["cdc detect checksum corrupted"] [schema=asim] [table=t2]
[2024/03/01 13:40:41.270 +00:00] [INFO] [message.go:461] ["data corrupted, print each column for debugging"] [name=pk] [type=15] [charset=utf8mb4] [flag=0] [value=105415910] [default=null]
[2024/03/01 13:40:41.271 +00:00] [INFO] [message.go:461] ["data corrupted, print each column for debugging"] [name=sk] [type=15] [charset=utf8mb4] [flag=0] [value=j1_24] [default=null]
[2024/03/01 13:40:41.271 +00:00] [INFO] [message.go:461] ["data corrupted, print each column for debugging"] [name=ts] [type=7] [charset=binary] [flag=0] [value="2024-03-01 13:40:39.662659"] [default=null]
[2024/03/01 13:40:41.271 +00:00] [INFO] [message.go:461] ["data corrupted, print each column for debugging"] [name=v] [type=15] [charset=utf8mb4] [flag=0] [value=cKpfKvsf7TyhGb5veZTAyATg4r7OFGSkwZeqM5ldPcvEVk6m6wHKbYg5UmNCJMT6Nybq9CFgRAUqRdBeNYhGmWU5tlL6gU3YYjE4a2ZtrDueDMui0HPLfS4toGz4WNQfhgqwcgWA6gJyaYmErL9BoAS21GzwehFmVT2WxDESLcPxG3TKcmDyY2K037pcNIvQcwubw6gTgv57J1F2rdda2EGXf9n1vAA0riLs5exlmWPVHlbaFxDZ6erwWIh5aNFOC3piWRArxYWTUqBq5pePqm9tUeZFJpZ9Uz8Ltn85EQZjAUvl0UoIYVF74wntqD3G0BMY9P0uir6zd7R9BAMPkmFKPPdc9IQiH0gCaOr3iKKOPji1B6bsTT7tvwnmnxPeSgOD4wHYqYcMbJ0AltrYhNDLPq3KqUIaAyd3Ba7zuJHXDge3eVbC64VqzLSjF5wwaqVKHMwykwrRB6xtt1an0u3OsC1bf7pyrZ6OG5q8TrP89KmrrKxXOytQqSXSGmQrOXuYQFaFYl6FKxcp9c7uKbD3iUsKad62ZgfQY2W238IhphdLznqQQfMNtzNUXRffR4XXBTXvN3giK4db5rxPU8jEptjsxphrI7UlVv9gUfNukGOitUUOlNQ2CSjqf7P43oY4diZr7yAKXKblkUVIulwd2oKJtiq9MjLy1enrr61r9xFIw5J10R0Ll5QdLkkaXylADB6Qey0rurNUdfg9jpHNjEoqmevGbsdAezQkKzCfFpa6BiGnlqcv7TRpHp2QKS5bCGuF2TZwbYv7dGbReSppbq2QhyixAc1oZ10O8Ki9NvBhianYJtFZKQ5yAQgzf61861bpcfvp5sbr88vT1WEoc9WCfEtFQag7BzQIreKYK473i8r6nVO1FYPgRuFsV7CDMCFY8zfLigLFtCfLykxoRxohK3VCxw5ZMJ9lx0BVdOOq72VZPJwThYvbwlG0hUADMMfM6UV7EGlqO7DZVRgy7s6ztkkaAgrojPDkqhVEaOPLzLpUs2mtXiGqHGKYbLpUR3LKQxIqKUb1JqNTJX7xD6e9u1KfM321FxVhR8xUOxXPqds854puIN4UuOJtdbhO4LdOFBCuU8gaJ0alItVGYFgZXRQT1JvHTIEeaxprBT8acKS2kfiACniaAIgjcrjN7zVIpcfM8zNYwdDSgiqQXszY18uXTo5kat4f0oh7AXnVKgNjFxcRbelvDPl6QDWacnaA3YlMvjFWULtPlVTrKLEaJP1kOQb6VvvPOavJYKtiujuSbmHgp4KufjSCFi4dnrltCveneh0U0Fd8C0o9uGj3u7oq938iMEMMPBfnLRwTq6FX4ECrp16ijXtGs9zn9cGWFWN3XhGyVoCdC5WBBFeF0bFRZ1PoElsABYF2udAMDuldEwpI6ykQwHnMlM4o4gh8ApfyWRsM4E3BYcPa7UtxjEZtIRFKKRfiAfD4gzs5AvpUgIhnkdF1X601AUZls8jgrPFSrQuec04RgkVU8Deq8zG1ojAXSgNuZLn8Kz296Wv3ocCCenEac8j7pz419JHE8nyONh59CSwA6Y6TOJvCx1cGKv5nOqoFuqM5vucSK9SwVG9wTMZvkDAo0cWtUjGfcRIYuwn1gyRS8s02RIruEOK8CScx65j9GJGnGiJNpeLhBKAsZJKrTWPqw0NCKaVsW8mxIvibtx0Tvw7INWAZRBlWmR0EdJj8QAJkqyF5rZF4NKdmJAmt8hDGpxlT1z6GUde6AevMgS6xFMTvARHOrEQZqOUS2zdpfluPgw136ESXY4XgDeiGueby47p3R3cJ0sMLl4doytwv89WHZJfRjsnfsLl6KUgs0btVOHa6AW5XSeis6ViQdt0BX3M12nW5cHBNRzc4MvSvkzsdMbavgW2R3Tic47S4S9GGfdqR3j0GUbR1lEMxKVnHfOUNvoRhiB8dMJtkUKVBEeltLjFsh6cFZtetLK0tAzGywLeq6YwZZwfN5UQb3BFoc7R9Nrt0C71erNGe9R6yznUe9Bw1Nl06FHYZI4DH26slDAAP09vnxhXj5JxFkJdzm12bUeJMHPmASvqZB5PJ2IuMSFaTR34yjcIEf5Xqft22Os42dTkKNZMrQkBwRjtef5XZhGHERTkPNAcwn8PpTdfkNaewPKe1fXVuBct20aZCYcmKgLHmg2UMU1fJsHu6TPtDcVmrN5d9GKN3PCInoGMkK4kuQqRg3sFEcWluapM5gsuH9hXYPmCi3MNXi1UJjp8fjneTD6Y88jOVRDOhGcPvxE5MqGlPAGlSuMoF0ZUg66NFayKsBe1rmDYTGEIgd4JhRSidzwg1nx83SLIsrZqxiSyDiw5lL2Av2QlzobMVa9B2cQlrx703rpj4yOyOSvWbshT6o5O1Tp3PctJ633iIS3Bh4f5IfrQIxlRGlr8a9Su2i7a2Brgje9y8Tus75pOU876HplUVqlyWgxqivxSrU47PxCzARnXSvAg23jziLRWsinaDEDxjtvUQ5MQpKYIq7CroNA5XyiqJ7MnmTBN2ltDmmHkuze4TRRecSqJO1XTpKVYdJ6X6NYx71HxsIK5bUEseJeypHQlSGEebz1xmXBTjixFmmEGWPV2Fj2YIZ6CrFdGWvzZlMEJOefvklDDjEoxlZL78Xmg4IccIQXQvaDkolFTNBrr6C2C82KoAilSy6Kb63sVsxdEMDrKKN2s8s8pr2HHtS55aTmdbcl98UvEO34VCHk0APHenTAojgMPnNIGb7nPoyDCkfzMmjFJqTQfYrF7h32z3M5xrW6QS53iOeqbBL7PvGm6InHjY3gXTgYsvj4bx91aYklYMbEiybjej9WnTGHVIXfaYPgJbxzgnGrTu5WVBmAM9Mcpf7S8WsxytD9YnVaK9WvImCp9kQrkYYMQFVEJnN3M7IYJzBUlJobrYNopBqY582aaHl3SdAFSRMnkJJvnZu63WeVPXGhfpTb038FvzVMfLsK9V2biHP41zMonH0fiCSMdoOu7LHrODoGafTigYgBQc7xHRNgZIyYM2JqTPxbGWwzw6oujDLGsFHtQhQ5uHR8s9odgEBXX7Ut9K3uVHa3TgUihePdSemB9rqJLg28iVjBl6zCHJqmL9FiIB0fXENetlalycx4Xnt0cYyZHUhXUDCW8yMsikpVaGRdSgCSVZ3aUKr9AGkU24ypi4ag7I7cx5yTSf8lL3cilqo7uxJSv5PilV9uX1p8ArHwVSyLYnn8XbSZu4UNDUzC8HdjbJanlpBFlWm90SZ5fpmXcb21T9BPpOumvyz0FnuAqSfhtr2sb6IuUKT49FpF92PKx2fSqFTlzFNqHBZDVZBfPWK0Ntbc8Kv0fiugUVExapoODBMWrli7jr362UrxmDka9L5TD39hC4W6xpKaIRrtlFPTxubpcpUgQyWiwcrWgXe05dV8xS7eIe8M9kQVlaDJjyi2GuCeOe5aylOVGCaKDv8NjfXwUg9d8Ln9TgoFRPma6ttI2pZcTtOyU0f6VoXzz8EWeCXVkD6z8wfdROGWsDdgxhHE6t2H2uLHJXx0Hu95CoqiuIxt1hOG9eIyhCTodKJafYx6iEY8al39HusfAxxbHFmddtaQGSWEWOwdhTasje75FHZfHRGFVsv1X7jEAF3Am5BdKuCgQMNLAgkaMFvaWzkGuyXDE5kqsmIfLZqbgILoTopSYIcC00VZL84EXkAX1wAAejXo1JpZdsgbeVyUaGybp5r0M3RqtKQNNerFk4NvvTFLMj0uY2n6STqGp190Xuy0QJvbEH4Tr6aEqv9vOB2frqw78FWCxpaPSwYCkPZST6TJzD4FX1LV533m00eu4aJi9Z7nk4g0QEDko1xJ26wAeW0Z9J6s4enVK32VHqPoMcBc4kS7Hq3UyEhJezLwm6nG8WLbkjd3qIC3bL7MwcI6t86I3OCBWnjrOkxSglKUUI5f4JKPnOYZElMsso8o6IsdkLTQLW7NdI1edkHUtEkX7GQIQgLt2CossDixvWCcbZO6tW5z9V6TP3OwaMGsRPbXK5kuGqOYvsZRUoBmeVbFo8mnYwAGyxygn8dJOw0ZCD8Hbe2YfAadK9HDFmHCpTGM3XhdpSBdzdbJcRoLHAuaupO8TTLrrpXR1sivtEpNxnn2MxUvGumEdQnIZ8a7ImfnxPWV0ufzm3PgOGLWBRtuElm8FbW5ZTEEeFMpB8NXZTLn1JCXldOHihfFK785r71UotoWwuzPh2P8LcTxDLanvw0M6IM6U0Mb2a8f5dHMXpD5fGaCg5yB3UoShdHtlShN20tmPw68MckwjZQVwBW8yBc7dC1VGalk7HJnBL13DxksuR7ixUXG86UBxVlg3BH4iLbs2B5lGHHEJruY3gcugF0CpcAxfNhErks0iO1ReJZp2VG9OdpIzk3LULHx6y2yiaMGEAaXGRTVrwdUUwzjBUCJHwrVeeFc17OkYwbp2GPzdwGjWSf0d0JS3U1grDeYkeMZX2SVaHYp6a7qoZ24hAukTwmmJnPrMZYTB6zW1DDJLFj2WdAPY5XTWO9vYnTqrZt8FpqrP7PLjZkpRdTaPc2Hqi7MT492lzuapBGdHtfdiRPnV7c6WmuuUXJuSTPdn5vpvV60sj7S5sasU8upynYqNxoRCZWp8d7IAwebshicibMyU2IpLMsEVqilNyqwttDCSHf6OE6uWnfzNFnh0C7HPFhJJdKqq8qrVnrssXq7ygEKaYMRbd8D2at59hNJgtcQX5zdM0n7q7amOBBLRM5KBjLEITCahKSu7FzBPU4P7vrtuy6LFKDQ53H8nwlcolMfkgLCVt7h1R8NYVyvA22ElgTcAYlXOuSCLycD6GeF30kPSbUC5Meja9nymJGK4d7qUZRqiqKVgmuoqxYi53jKkdy41sXzYAMyvuyRMP9y46y1b5SWiu2jcyeRxMpIQI1izCrRFOS6f0mRoFLBIfXpe6aGNJONE0O2uqnn055tjle7EOh2OG6jyPTRbmY2FS79YODP2e0oFxtsUbXtE9O13GJGskXQkpe5SWKk00bXZ5ZRUPHrxLmsFCJUzuO2U4qyj7r7hE6K30XbtNIOMuJAaxQRWPtVDPuv4GS3VeHHkRAT2a0HofrJ9WM43mCIeXbecQU33ubguKbX6AX438ZiKy66n1oQ2HewS0kE35SePJEaT7Y7AKyij2lSSJlr9Kvx2VyFsgHzJtjbs3QwBhopjNNx1PrTPo7wgTdfeaVupKRwXOh9Tx1Qk7I2xq57L9v85ZJr2cqhNIAujiT4xhNjyE94waGBdUOTnEOSFXPpZCRUy6udOltYEKrrrx5c2vGsDuuiIIgPt4YdaBC76GQNjvHjoqlbVMFRqRbyb7NvNogFHd721aRgKA80vTPwFmH1iIRWPhVKjxxf22zxWPJtuDibXiZjgZsMH0VajQXZBmmQhn8PPNXFHGrc958Dcdo9Iv6oRu3MI1uCxqEBKMwEEZvTY1QvRvu3UfsS6hN2fzgQUuxaPD1dRaxooahwD7lEtxJRtDJ9CuZWO43VFZtJvRgFDpUKmWYMmEXx3LwlsCzTIrZr4pVQbpE5WPf6Ydr9JExUBeMut2pbLVqaBSKzamiwWhH4mO6qOITaZD4QcygC4S1Nh3fOpuyWbWwuax7mxc48lVDo5kU9XTVACU6GY1XlYPqtpNkeLrp17hTUGczBMB5RHm0M0g8hVBCsMWxtpUa9yBKBFZC8hPWrQ1y20uHuws6ENWmuXZrcyARtfCCi8N08eK1GHQKiuSt1jSlLPbeJYKRmoaLgFEoIjjRmHEFi4fg5TvDb53hIaxBibDVkAnW1ZeUOA2urzOdTLki1tzboUCHlug5mj1Ib5jihCl2TjDtCRpwjj9Q45enXfI1Xk46tlIG2bcFfIVZHCCroJOqvddPuaT0uvucSZAspTiRU7zvToIafgHNQeprOgoomC8Ow5CiIXTFcnrX56tA3gql1fM0jH28BX6khaSXbHs3J8k57CmQIc554QuO4OrzYIUX9L6yDIpOv65uM3JpWPVIdOIpjkhDqwdVaBX323JrpYtk5kK0NFOkrYSPL0dgv9qGQYvk4SjCSoup5ANG12Ht9p2417SPGpn92iWUGE46ZSL17W6ektlkeHSotlx97JrJwgy6WsWwiCC2rOiOmnuelmMAkmMwg8bIM5BDoTuvrFtJknNii0AYrvLNWsvXbAGQ4WRwFmFkkueTqPJcO2snQtdYh55KoMjWkKelWvrw6c3LntkG5Serc6NyyWil5XAlEIxxSMTSlrybJxM8uYeEyTxLKfLq55vWlfaplcfQ6BoGSJ687OZ4wxhXeoVetGoZnwPbw5575cFTbCMJnsIQlCuLAu2aAH6KQ3ZwO7RWpcpHisawJsRqUwvljyxOkynIT6UYS4NR3BULxe5YruTNEI6H3iBSGcZEJaE135NJD4RpTaEKPYusL82IzOxdU0phYaiz2uIfU78LETWqWcmZPH5Eb0dgzUxOgm4j3TZHdtkCPfeNayAK4OvnAx5ilNR5habIperDtQl1jj1zXXadXl3UdUOxQV3H20fNTX1v9fzFvttL6Q98pHyr] [default=null]
[2024/03/01 13:40:41.271 +00:00] [PANIC] [main.go:662] ["decode message value failed"] [value="\u0016\u0006\u0006\u0002\u0012asim\u0004t2\ufffd\r\u0000\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\u000c؂\ufffd\ufffd\ufffdc\ufffd\ufffd\ufffd\ufffdǐ\ufffd\ufffd\u000c\u0000\u0000\u0002\u0000\u0000\ufffd\ufffd\ufffd\ufffd\u0005\u0000\u0002\u0008\u0004pk\u0008\u0012105415910\u0004sk\u0008\nj1_24\u0004ts\u000c\u0006UTC42024-03-01 13:40:39.662659\u0002v\u0008\ufffdfcKpfKvsf7TyhGb5veZTAyATg4r7OFGSkwZeqM5ldPcvEVk6m6wHKbYg5UmNCJMT6Nybq9CFgRAUqRdBeNYhGmWU5tlL6gU3YYjE4a2ZtrDueDMui0HPLfS4toGz4WNQfhgqwcgWA6gJyaYmErL9BoAS21GzwehFmVT2WxDESLcPxG3TKcmDyY2K037pcNIvQcwubw6gTgv57J1F2rdda2EGXf9n1vAA0riLs5exlmWPVHlbaFxDZ6erwWIh5aNFOC3piWRArxYWTUqBq5pePqm9tUeZFJpZ9Uz8Ltn85EQZjAUvl0UoIYVF74wntqD3G0BMY9P0uir6zd7R9BAMPkmFKPPdc9IQiH0gCaOr3iKKOPji1B6bsTT7tvwnmnxPeSgOD4wHYqYcMbJ0AltrYhNDLPq3KqUIaAyd3Ba7zuJHXDge3eVbC64VqzLSjF5wwaqVKHMwykwrRB6xtt1an0u3OsC1bf7pyrZ6OG5q8TrP89KmrrKxXOytQqSXSGmQrOXuYQFaFYl6FKxcp9c7uKbD3iUsKad62ZgfQY2W238IhphdLznqQQfMNtzNUXRffR4XXBTXvN3giK4db5rxPU8jEptjsxphrI7UlVv9gUfNukGOitUUOlNQ2CSjqf7P43oY4diZr7yAKXKblkUVIulwd2oKJtiq9MjLy1enrr61r9xFIw5J10R0Ll5QdLkkaXylADB6Qey0rurNUdfg9jpHNjEoqmevGbsdAezQkKzCfFpa6BiGnlqcv7TRpHp2QKS5bCGuF2TZwbYv7dGbReSppbq2QhyixAc1oZ10O8Ki9NvBhianYJtFZKQ5yAQgzf61861bpcfvp5sbr88vT1WEoc9WCfEtFQag7BzQIreKYK473i8r6nVO1FYPgRuFsV7CDMCFY8zfLigLFtCfLykxoRxohK3VCxw5ZMJ9lx0BVdOOq72VZPJwThYvbwlG0hUADMMfM6UV7EGlqO7DZVRgy7s6ztkkaAgrojPDkqhVEaOPLzLpUs2mtXiGqHGKYbLpUR3LKQxIqKUb1JqNTJX7xD6e9u1KfM321FxVhR8xUOxXPqds854puIN4UuOJtdbhO4LdOFBCuU8gaJ0alItVGYFgZXRQT1JvHTIEeaxprBT8acKS2kfiACniaAIgjcrjN7zVIpcfM8zNYwdDSgiqQXszY18uXTo5kat4f0oh7AXnVKgNjFxcRbelvDPl6QDWacnaA3YlMvjFWULtPlVTrKLEaJP1kOQb6VvvPOavJYKtiujuSbmHgp4KufjSCFi4dnrltCveneh0U0Fd8C0o9uGj3u7oq938iMEMMPBfnLRwTq6FX4ECrp16ijXtGs9zn9cGWFWN3XhGyVoCdC5WBBFeF0bFRZ1PoElsABYF2udAMDuldEwpI6ykQwHnMlM4o4gh8ApfyWRsM4E3BYcPa7UtxjEZtIRFKKRfiAfD4gzs5AvpUgIhnkdF1X601AUZls8jgrPFSrQuec04RgkVU8Deq8zG1ojAXSgNuZLn8Kz296Wv3ocCCenEac8j7pz419JHE8nyONh59CSwA6Y6TOJvCx1cGKv5nOqoFuqM5vucSK9SwVG9wTMZvkDAo0cWtUjGfcRIYuwn1gyRS8s02RIruEOK8CScx65j9GJGnGiJNpeLhBKAsZJKrTWPqw0NCKaVsW8mxIvibtx0Tvw7INWAZRBlWmR0EdJj8QAJkqyF5rZF4NKdmJAmt8hDGpxlT1z6GUde6AevMgS6xFMTvARHOrEQZqOUS2zdpfluPgw136ESXY4XgDeiGueby47p3R3cJ0sMLl4doytwv89WHZJfRjsnfsLl6KUgs0btVOHa6AW5XSeis6ViQdt0BX3M12nW5cHBNRzc4MvSvkzsdMbavgW2R3Tic47S4S9GGfdqR3j0GUbR1lEMxKVnHfOUNvoRhiB8dMJtkUKVBEeltLjFsh6cFZtetLK0tAzGywLeq6YwZZwfN5UQb3BFoc7R9Nrt0C71erNGe9R6yznUe9Bw1Nl06FHYZI4DH26slDAAP09vnxhXj5JxFkJdzm12bUeJMHPmASvqZB5PJ2IuMSFaTR34yjcIEf5Xqft22Os42dTkKNZMrQkBwRjtef5XZhGHERTkPNAcwn8PpTdfkNaewPKe1fXVuBct20aZCYcmKgLHmg2UMU1fJsHu6TPtDcVmrN5d9GKN3PCInoGMkK4kuQqRg3sFEcWluapM5gsuH9hXYPmCi3MNXi1UJjp8fjneTD6Y88jOVRDOhGcPvxE5MqGlPAGlSuMoF0ZUg66NFayKsBe1rmDYTGEIgd4JhRSidzwg1nx83SLIsrZqxiSyDiw5lL2Av2QlzobMVa9B2cQlrx703rpj4yOyOSvWbshT6o5O1Tp3PctJ633iIS3Bh4f5IfrQIxlRGlr8a9Su2i7a2Brgje9y8Tus75pOU876HplUVqlyWgxqivxSrU47PxCzARnXSvAg23jziLRWsinaDEDxjtvUQ5MQpKYIq7CroNA5XyiqJ7MnmTBN2ltDmmHkuze4TRRecSqJO1XTpKVYdJ6X6NYx71HxsIK5bUEseJeypHQlSGEebz1xmXBTjixFmmEGWPV2Fj2YIZ6CrFdGWvzZlMEJOefvklDDjEoxlZL78Xmg4IccIQXQvaDkolFTNBrr6C2C82KoAilSy6Kb63sVsxdEMDrKKN2s8s8pr2HHtS55aTmdbcl98UvEO34VCHk0APHenTAojgMPnNIGb7nPoyDCkfzMmjFJqTQfYrF7h32z3M5xrW6QS53iOeqbBL7PvGm6InHjY3gXTgYsvj4bx91aYklYMbEiybjej9WnTGHVIXfaYPgJbxzgnGrTu5WVBmAM9Mcpf7S8WsxytD9YnVaK9WvImCp9kQrkYYMQFVEJnN3M7IYJzBUlJobrYNopBqY582aaHl3SdAFSRMnkJJvnZu63WeVPXGhfpTb038FvzVMfLsK9V2biHP41zMonH0fiCSMdoOu7LHrODoGafTigYgBQc7xHRNgZIyYM2JqTPxbGWwzw6oujDLGsFHtQhQ5uHR8s9odgEBXX7Ut9K3uVHa3TgUihePdSemB9rqJLg28iVjBl6zCHJqmL9FiIB0fXENetlalycx4Xnt0cYyZHUhXUDCW8yMsikpVaGRdSgCSVZ3aUKr9AGkU24ypi4ag7I7cx5yTSf8lL3cilqo7uxJSv5PilV9uX1p8ArHwVSyLYnn8XbSZu4UNDUzC8HdjbJanlpBFlWm90SZ5fpmXcb21T9BPpOumvyz0FnuAqSfhtr2sb6IuUKT49FpF92PKx2fSqFTlzFNqHBZDVZBfPWK0Ntbc8Kv0fiugUVExapoODBMWrli7jr362UrxmDka9L5TD39hC4W6xpKaIRrtlFPTxubpcpUgQyWiwcrWgXe05dV8xS7eIe8M9kQVlaDJjyi2GuCeOe5aylOVGCaKDv8NjfXwUg9d8Ln9TgoFRPma6ttI2pZcTtOyU0f6VoXzz8EWeCXVkD6z8wfdROGWsDdgxhHE6t2H2uLHJXx0Hu95CoqiuIxt1hOG9eIyhCTodKJafYx6iEY8al39HusfAxxbHFmddtaQGSWEWOwdhTasje75FHZfHRGFVsv1X7jEAF3Am5BdKuCgQMNLAgkaMFvaWzkGuyXDE5kqsmIfLZqbgILoTopSYIcC00VZL84EXkAX1wAAejXo1JpZdsgbeVyUaGybp5r0M3RqtKQNNerFk4NvvTFLMj0uY2n6STqGp190Xuy0QJvbEH4Tr6aEqv9vOB2frqw78FWCxpaPSwYCkPZST6TJzD4FX1LV533m00eu4aJi9Z7nk4g0QEDko1xJ26wAeW0Z9J6s4enVK32VHqPoMcBc4kS7Hq3UyEhJezLwm6nG8WLbkjd3qIC3bL7MwcI6t86I3OCBWnjrOkxSglKUUI5f4JKPnOYZElMsso8o6IsdkLTQLW7NdI1edkHUtEkX7GQIQgLt2CossDixvWCcbZO6tW5z9V6TP3OwaMGsRPbXK5kuGqOYvsZRUoBmeVbFo8mnYwAGyxygn8dJOw0ZCD8Hbe2YfAadK9HDFmHCpTGM3XhdpSBdzdbJcRoLHAuaupO8TTLrrpXR1sivtEpNxnn2MxUvGumEdQnIZ8a7ImfnxPWV0ufzm3PgOGLWBRtuElm8FbW5ZTEEeFMpB8NXZTLn1JCXldOHihfFK785r71UotoWwuzPh2P8LcTxDLanvw0M6IM6U0Mb2a8f5dHMXpD5fGaCg5yB3UoShdHtlShN20tmPw68MckwjZQVwBW8yBc7dC1VGalk7HJnBL13DxksuR7ixUXG86UBxVlg3BH4iLbs2B5lGHHEJruY3gcugF0CpcAxfNhErks0iO1ReJZp2VG9OdpIzk3LULHx6y2yiaMGEAaXGRTVrwdUUwzjBUCJHwrVeeFc17OkYwbp2GPzdwGjWSf0d0JS3U1grDeYkeMZX2SVaHYp6a7qoZ24hAukTwmmJnPrMZYTB6zW1DDJLFj2WdAPY5XTWO9vYnTqrZt8FpqrP7PLjZkpRdTaPc2Hqi7MT492lzuapBGdHtfdiRPnV7c6WmuuUXJuSTPdn5vpvV60sj7S5sasU8upynYqNxoRCZWp8d7IAwebshicibMyU2IpLMsEVqilNyqwttDCSHf6OE6uWnfzNFnh0C7HPFhJJdKqq8qrVnrssXq7ygEKaYMRbd8D2at59hNJgtcQX5zdM0n7q7amOBBLRM5KBjLEITCahKSu7FzBPU4P7vrtuy6LFKDQ53H8nwlcolMfkgLCVt7h1R8NYVyvA22ElgTcAYlXOuSCLycD6GeF30kPSbUC5Meja9nymJGK4d7qUZRqiqKVgmuoqxYi53jKkdy41sXzYAMyvuyRMP9y46y1b5SWiu2jcyeRxMpIQI1izCrRFOS6f0mRoFLBIfXpe6aGNJONE0O2uqnn055tjle7EOh2OG6jyPTRbmY2FS79YODP2e0oFxtsUbXtE9O13GJGskXQkpe5SWKk00bXZ5ZRUPHrxLmsFCJUzuO2U4qyj7r7hE6K30XbtNIOMuJAaxQRWPtVDPuv4GS3VeHHkRAT2a0HofrJ9WM43mCIeXbecQU33ubguKbX6AX438ZiKy66n1oQ2HewS0kE35SePJEaT7Y7AKyij2lSSJlr9Kvx2VyFsgHzJtjbs3QwBhopjNNx1PrTPo7wgTdfeaVupKRwXOh9Tx1Qk7I2xq57L9v85ZJr2cqhNIAujiT4xhNjyE94waGBdUOTnEOSFXPpZCRUy6udOltYEKrrrx5c2vGsDuuiIIgPt4YdaBC76GQNjvHjoqlbVMFRqRbyb7NvNogFHd721aRgKA80vTPwFmH1iIRWPhVKjxxf22zxWPJtuDibXiZjgZsMH0VajQXZBmmQhn8PPNXFHGrc958Dcdo9Iv6oRu3MI1uCxqEBKMwEEZvTY1QvRvu3UfsS6hN2fzgQUuxaPD1dRaxooahwD7lEtxJRtDJ9CuZWO43VFZtJvRgFDpUKmWYMmEXx3LwlsCzTIrZr4pVQbpE5WPf6Ydr9JExUBeMut2pbLVqaBSKzamiwWhH4mO6qOITaZD4QcygC4S1Nh3fOpuyWbWwuax7mxc48lVDo5kU9XTVACU6GY1XlYPqtpNkeLrp17hTUGczBMB5RHm0M0g8hVBCsMWxtpUa9yBKBFZC8hPWrQ1y20uHuws6ENWmuXZrcyARtfCCi8N08eK1GHQKiuSt1jSlLPbeJYKRmoaLgFEoIjjRmHEFi4fg5TvDb53hIaxBibDVkAnW1ZeUOA2urzOdTLki1tzboUCHlug5mj1Ib5jihCl2TjDtCRpwjj9Q45enXfI1Xk46tlIG2bcFfIVZHCCroJOqvddPuaT0uvucSZAspTiRU7zvToIafgHNQeprOgoomC8Ow5CiIXTFcnrX56tA3gql1fM0jH28BX6khaSXbHs3J8k57CmQIc554QuO4OrzYIUX9L6yDIpOv65uM3JpWPVIdOIpjkhDqwdVaBX323JrpYtk5kK0NFOkrYSPL0dgv9qGQYvk4SjCSoup5ANG12Ht9p2417SPGpn92iWUGE46ZSL17W6ektlkeHSotlx97JrJwgy6WsWwiCC2rOiOmnuelmMAkmMwg8bIM5BDoTuvrFtJknNii0AYrvLNWsvXbAGQ4WRwFmFkkueTqPJcO2snQtdYh55KoMjWkKelWvrw6c3LntkG5Serc6NyyWil5XAlEIxxSMTSlrybJxM8uYeEyTxLKfLq55vWlfaplcfQ6BoGSJ687OZ4wxhXeoVetGoZnwPbw5575cFTbCMJnsIQlCuLAu2aAH6KQ3ZwO7RWpcpHisawJsRqUwvljyxOkynIT6UYS4NR3BULxe5YruTNEI6H3iBSGcZEJaE135NJD4RpTaEKPYusL82IzOxdU0phYaiz2uIfU78LETWqWcmZPH5Eb0dgzUxOgm4j3TZHdtkCPfeNayAK4OvnAx5ilNR5habIperDtQl1jj1zXXadXl3UdUOxQV3H20fNTX1v9fzFvttL6Q98pHyr\u0000\u0000"] [error="[CDC:ErrDecodeFailed]decode failed: checksum corrupted"] [errorVerbose="[CDC:ErrDecodeFailed]decode failed: checksum corrupted\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20231212100244-799fae176cfb/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20231212100244-799fae176cfb/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/sink/codec/simple.buildRowChangedEvent\n\tgithub.com/pingcap/tiflow/pkg/sink/codec/simple/message.go:469\ngithub.com/pingcap/tiflow/pkg/sink/codec/simple.(*Decoder).NextRowChangedEvent\n\tgithub.com/pingcap/tiflow/pkg/sink/codec/simple/decoder.go:167\nmain.(*Consumer).ConsumeClaim\n\tgithub.com/pingcap/tiflow/cmd/kafka-consumer/main.go:660\ngithub.com/IBM/sarama.(*consumerGroupSession).consume\n\tgithub.com/IBM/sarama@v1.41.2/consumer_group.go:949\ngithub.com/IBM/sarama.newConsumerGroupSession.func2\n\tgithub.com/IBM/sarama@v1.41.2/consumer_group.go:874\nruntime.goexit\n\truntime/asm_amd64.s:1650"] [stack="main.(*Consumer).ConsumeClaim\n\tgithub.com/pingcap/tiflow/cmd/kafka-consumer/main.go:662\ngithub.com/IBM/sarama.(*consumerGroupSession).consume\n\tgithub.com/IBM/sarama@v1.41.2/consumer_group.go:949\ngithub.com/IBM/sarama.newConsumerGroupSession.func2\n\tgithub.com/IBM/sarama@v1.41.2/consumer_group.go:874"]
```

### Versions of the cluster

TiCDC version: master
Kafka consumer version: master

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #10703, close #10705

### What is changed and how it works?

* timestamp parse format with microsecond precision.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: pkg/sink/codec/common/verify_checksum.go [hunk 1]
```diff
@@ -154,16 +155,20 @@ func buildChecksumBytes(buf []byte, value interface{}, mysqlType byte) ([]byte,
 	case mysql.TypeTimestamp:
 		location := config.GetDefaultServerConfig().TZ
 		timestamp := value.(string)
+		format := "2006-01-02 15:04:05"
```
Comment (asddongmen) [inline pkg/sink/codec/common/verify_checksum.go:158]: Could you please add some comments or reference information about why should do these check and conversion?

**Review by asddongmen — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Pull request merged**